src/pages/faq/comparison/terraform.mdx

```diff
@@ -0,0 +1,155 @@
+
+<CodeGroup compare>
+
+```javascript {{title: 'Nitric', label: 'handle.ts' }}
+import * as nitric from '@nitric/sdk'
+
+const bucket = nitric.bucket('my-bucket').for('reading', 'writing')
+
+bucket.on('create', '*', async (ctx) => {
+  console.log(ctx.file.key, 'was created')
+})
+```
+
+```hcl {{title: 'Terraform', label: 'main.tf'}}
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 3.0"
+    }
+  }
+}
+
+provider "aws" {
+  region = "us-west-2"
+}
+
+locals {
+  lambda_function_name = "upload_hello_txt_lambda"
+}
+
+resource "aws_s3_bucket" "this" {
+  bucket = "my-s3-bucket"
+  acl    = "private"
+}
+
+data "archive_file" "lambda_zip" {
+  type        = "zip"
+  source_file = "index.js"
+  output_path = "${path.module}/lambda.zip"
+}
+
+resource "aws_lambda_function" "this" {
+  function_name = local.lambda_function_name
+  role          = aws_iam_role.lambda_role.arn
+  handler       = "index.handler"
+  runtime       = "nodejs14.x"
+  filename      = data.archive_file.lambda_zip.output_path
+  timeout       = 10
+
+  environment {
+    variables = {
+      BUCKET_NAME = aws_s3_bucket.this.bucket
+    }
+  }
+}
+
+resource "aws_iam_role" "lambda_role" {
+  name = "lambda_role"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = "sts:AssumeRole"
+        Effect = "Allow"
+        Principal = {
+          Service = "lambda.amazonaws.com"
+        }
+      }
+    ]
+  })
+}
+
+resource "aws_iam_role_policy" "lambda_policy" {
+  name = "lambda_policy"
+  role = aws_iam_role.lambda_role.id
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = [
+          "logs:CreateLogGroup",
+          "logs:CreateLogStream",
+          "logs:PutLogEvents"
+        ]
+        Effect   = "Allow"
+        Resource = "arn:aws:logs:*:*:*"
+      },
+      {
+        Action = [
+          "s3:PutObject"
+        ]
+        Effect   = "Allow"
+        Resource = "${aws_s3_bucket.this.arn}/*"
+      }
+    ]
+  })
+}
+
+output "bucket_name" {
+  value = aws_s3_bucket.this.bucket
+}
+
+output "lambda_function_name" {
+  value = aws_lambda_function.this.function_name
+}
+```
+
+```javascript {{title: 'Terraform', label: 'src/index.js'}}
+const AWS = require('aws-sdk')
+
+exports.handler = async (event) => {
+  const {
+    s3: { object },
+  } = event
+  console.log(object.key, 'was created')
+}
+```
+
+</CodeGroup>
+
+> In the pulumi example we get control over absolutely every facet of our infrastructure which is great, but for the majority of use cases a repeatable pattern of deployment is enough and scales much better than applying fine grained configuration every time we need to define a new deployment (with Pulumi this would typically be done by sharing infrastructure as libraries e.g. Pulumi Crosswalks).
```

Comment:
```suggestion
> In the Pulumi example we get control over absolutely every facet of our infrastructure which is great, but for the majority of use cases, a repeatable pattern of deployment is enough and scales much better than applying fine-grained configuration every time we need to define a new deployment (with Pulumi this would typically be done by sharing infrastructure as libraries e.g. Pulumi Crosswalks).
```

Reply:
Shouldnt it be terraform?